server/src/main/java/io/crate/types/IntervalType.java

```diff
@@ -181,12 +182,29 @@ public long valueBytes(Period value) {
         return 32;
     }
 
+    /**
+     * Copy of {@link Period#toStandardDuration()}, but also includes 30 days per month and
+     * 12 months per year calculation. Uses BigInteger to avoid long overflow in case of large
+     * numbers.
+     */
+    public static BigInteger toStandardDuration(Period p) {
+        BigInteger millis = BigInteger.valueOf(p.getMillis());
+        millis = millis.add(BigInteger.valueOf(p.getSeconds() * (long) DateTimeConstants.MILLIS_PER_SECOND));
+        millis = millis.add(BigInteger.valueOf(p.getMinutes() * (long) DateTimeConstants.MILLIS_PER_MINUTE));
+        millis = millis.add(BigInteger.valueOf(p.getHours() * (long) DateTimeConstants.MILLIS_PER_HOUR));
+        millis = millis.add(BigInteger.valueOf(p.getDays() * (long) DateTimeConstants.MILLIS_PER_DAY));
+        millis = millis.add(BigInteger.valueOf(p.getWeeks() * (long) DateTimeConstants.MILLIS_PER_WEEK));
+        millis = millis.add(BigInteger.valueOf(p.getMonths() * 30 * (long) DateTimeConstants.MILLIS_PER_DAY));
+        millis = millis.add(BigInteger.valueOf(p.getYears() * 12 * 30 * (long) DateTimeConstants.MILLIS_PER_DAY));
+        return millis;
+    }
+
```

Comment:
Minor:
 
 I think we can slightly reduce number of allocations per addition:
 
 `Period.get***` methods, even `getMillis` return `int`. 
 Mills in a week is ~10^8 (makes it 10^17 when multiplied by int), so up to weeks we are fine with long:
 to overflow we need 100 more 10^17 magnitude summands but all units up to weeks are much smaller because DateTimeConstants values are small.
 
We can use our `OverflowAwareMutableLong` which has logic "use primitive as much as possible, flush when needed to BigDecimal, then reset and again start with primitives".  Since it's BigDecimal, we would need OverflowAwareMutableLong.value().toBigInteger() at the end but it will save us those 5 allocations

Reply:
Thx @mfussenegger! I will look more into this, and see if I can follow up with improvement.